spec/features/licence_transaction_spec.rb

```diff
@@ -0,0 +1,57 @@
+require "spec_helper"
+
+RSpec.feature "Creating a Licence", type: :feature do
+  let(:fields)              { %i[base_path content_id public_updated_at title publication_state] }
+  let(:licence_transaction) { FactoryBot.create(:licence_transaction) }
+  let(:content_id)          { licence_transaction["content_id"] }
+  let(:public_updated_at)   { licence_transaction["public_updated_at"] }
+
+  before do
+    log_in_as_editor(:licence_transaction_editor)
+    allow(SecureRandom).to receive(:uuid).and_return(content_id)
+
+    stub_publishing_api_has_content([licence_transaction], hash_including(document_type: LicenceTransaction.document_type))
+    stub_publishing_api_has_item(licence_transaction)
+    stub_any_publishing_api_put_content
+    stub_any_publishing_api_patch_links
+  end
+
+  scenario "adding a new licence" do
+    visit "/find-licences"
+    click_link "Add another Licence"
+    expect(page.status_code).to eq(200)
+    expect(page.current_path).to eq("/licences/new")
+  end
+
+  scenario "saving a new licence case with no data" do
+    visit "/licences/new"
+
+    click_button "Save as draft"
+
+    expect(page.status_code).to eq(422)
+
+    expect(page).to have_content("Please fix the following errors")
+    expect(page).to have_content("Title can't be blank")
+    expect(page).to have_content("Summary can't be blank")
+    expect(page).to have_content("Body can't be blank")
+  end
+
+  scenario "saving a new licence with valid data" do
+    visit "/licences/new"
+
+    fill_in "Title", with: "Example licence"
+    fill_in "Summary", with: "This is the summary of a licence"
+    fill_in "Body", with: "## Header#{"\n\nThis is the long body of a licence" * 2}"
+    select "England", from: "Country"
+    select "Education", from: "Sector"
+    select "Use CCTV systems", from: "Activity"
+    fill_in "Continuation link", with: "https://www.gov.uk"
+    fill_in "Will continue on", with: "on GOV.UK"
+
+    click_button "Save as draft"
+    assert_publishing_api_put_content(content_id)
```

Comment:
Maybe consider passing payload value here?

Reply:
Nice suggestion, I did ponder about testing the payload (like some other features) but decided against it for a couple of reasons:

1) It is a bit strange to be testing API calls / payload in a feature test, as the expectations should just be what appears on the page versus what the underlying code is doing (api calls, model updates etc). Feels like asserting a status and a generic call is kind of okay, but asserting a full payload seems too much.

2) The payload / data structure is already tested in other places, e.g the [model is validated against Publishing API's schema]( https://github.com/alphagov/specialist-publisher/pull/2169/files) and the [shared payload presenter has a spec]( https://github.com/alphagov/specialist-publisher/blob/main/spec/presenters/document_presenter_spec.rb).

I've also tested various scenarios against the feature test, like removing a model attribute, updating a field name etc which all results in the feature test failing so pretty confident we don't need to assert the payload too.

Reply:
Sounds reasonable 👍